chore(gulp) : convert grunt to gulp

fob413 opened the pull request:
#### What does this PR do?

Adds gulpfile as a preferable task runner to grunt

#### Description of Task to be completed?

Run all scripts from gulpfile and integrate coveralls

#### How should this be manually tested?

Run **gulp** from command line

#### Any background context you want to provide?

The app currently uses grunt as its task runner

#### What are the relevant pivotal tracker stories?

https://www.pivotaltracker.com/story/show/150398703

Commit:
chore(gulp) : convert grunt to gulp

- install necessary gulp dependencies
- setup gulp tasks
- change start script to use gulp
- delete gruntfile.js

[Delivers #150398703]

VanPaitin commented on package.json:
```diff
@@ -30,6 +31,21 @@
     "grunt": "~0.4.1",
     "grunt-bower-task": "~0.4.0",
     "grunt-cli": "~0.1.9",
```
Don't you think that we can do away with any grunt related dependencies

VanPaitin reviewed (changes requested):
Try and correct the `.eslintrc` file so hound will be able to parse it normally. I also dropped a comment concerning `grunt` dependencies

Commit:
chore(gulp): fix hound error

change eslint file from .eslintrc.json to .eslintrc

[Delivers #150398703]

Commit:
chore(gulp): fix hound error

correct all hound errors in files modified

[Delivers #150398703]

VanPaitin reviewed (changes requested):
There are still many hound issues. Please be very meticulous. Attention to detail is key. Try to resolve them. You might ignore the module resolution cop for now and address every other one. How come your when your run eslint on your local machine, it is not finding the errors

VanPaitin commented on gulpfile.babel.js:
```diff
@@ -0,0 +1,119 @@
+        [
+          '/test/user/model.js', 
+          'test/game/game.js'
+        ], 
```
Resolve trailing spaces

VanPaitin commented on gulpfile.babel.js:
```diff
@@ -0,0 +1,119 @@
+    .on('finish', () => {
+      gulp.src(
+        [
+          '/test/user/model.js', 
```
Resolve trailing spaces

Mcdavid95 committed:
chore(gulp): fix-hound-error

- turn off eslint resolve import rule
- turn off nodemon watch

[Delivers #150398703]

Mcdavid95 committed:
chore(gulp): fix-eslint-issues

- turn off commonjs rule

[Delivers #150398703]
